web/source/kmwuibutton.ts

```diff
@@ -213,15 +214,21 @@ if(!window['keyman']['ui']['name']) {
         } 
         else
         {
-          kbdId.className = osk['isEnabled']() ? 'kmw_show' : 'kmw_hide';
+          let osk = keymanweb.osk;
+          kbdId.className = osk && osk['isEnabled']() ? 'kmw_show' : 'kmw_hide';
         }
       }
     }  
   
-    /**
-     * UI Functions called by KeymanWeb or OSK
-     */     
-    osk['addEventListener']('show',
+    ui.registerEvents = function() {
+      let osk = keymanweb.osk;
+      if(!osk) {
+        return;
+      }
+      /**
+       * UI Functions called by KeymanWeb or OSK
+       */     
+      osk['addEventListener']('show',
       function(oskPosition)
```

Comment:
Indenting here looks a bit strange? Also on line 249-250.

Reply:
Welcome to the UI modules.  A PR simply to rectify all of their formatting issues would still be a chore to review.

Reply:
They haven't been _seriously_ maintained since before we went open-source.